docs/book/src/topics/container-networking-interface.md

```diff
@@ -0,0 +1,21 @@
+# Container Networking Interface
+
+This document aims to provide steps to configure your cluster using the below CNI solutions
+
+- [Azure CNI v1](#azure-container-networking-interface-v1)
+
+## Azure Container Networking Interface v1
+
+As of writing this document, Azure CNI needs to be installed in the following steps below. 
+
+- Bring up your workload cluster with desired number of Public IP Configs
+- Update Azure VNet Config in `/etc/cloud/cloud.cfg.d/` with the following contents placed in `15-azure-vnet.cfg` file
```

Comment:
If this file must be on a host, I think it should be added via `KubeadmControlPlane` `files` field instead of being added in image-builder, as installed CNI is runtime-specific. It could even be created by an init-container of `azure-cni` DaemonSet really.

Reply:
`15-azure-vnet.cfg` is read by the cloud-init and it is supposed to be read before the initial boot of the VM to setup network config of the VM instance.
I don't fully understand adding this file via `KubeadmControlPlane`  `files`.. Can we achieve adding this config file via `KubeadmControlPlane` to the VM before its initial boot? 
I would like to learn more about this option, can you please point me to it?

Reply:
So CAPI Bootstrap provider Kubeadm (CAPBK) is currently what's responsible for generating cloud-init for Azure nodes. Given that configuration from `/etc/cloud/cloud.cfg.d` is read by cloud-init, what we aim here is to configure cloud-init with `Azure.apply_network_config: false`, right? So perhaps CAPBK could generate such network configuration when requested by Azure cluster templates? Either via file if that works, or via `datasource`, which currently doesn't seem customizable. I'd try modifying the following code to try it out: https://github.com/kubernetes-sigs/cluster-api/blob/ad110582f9aee631e0d718fb0ccd9461b1a9d230/bootstrap/kubeadm/internal/cloudinit/controlplane_init.go#L17.

Reply:
> So perhaps CAPBK could generate such network configuration when requested by Azure cluster templates.

I am understanding this as a light coupling between CAPZ and Azure CNI, where CAPZ, while bringing up workload cluster, requests CABPK to init the VMs `Azure.apply_network_config: false`. If it is so, I don't think we would want any CNI related configs tied to CAPZ.

@CecileRobertMichon @mboersma please weigh in your thoughts too. 

----
> So perhaps CAPBK could generate such network configuration when requested by Azure cluster templates

On the other hand, could the above request be generated by Azure CNI templates?

Reply:
> On the other hand, could the above request be generated by Azure CNI templates?

I think this is what @invidian is suggesting, adding the file directly in the template under the Kubeadm config: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/cluster-template.yaml#L180

Reply:
resolving since https://github.com/kubernetes-sigs/image-builder/pull/1090 merged